Projects/UOContent/Mobiles/AI/BaseAI.cs

```diff
@@ -1834,242 +1760,274 @@ public virtual bool DoOrderTransfer()
         {
             if (m_Mobile.Debug)
             {
-                m_Mobile.DebugSay($"Begin transfer with {to.Name}");
+                m_Mobile.DebugSay($"Beginning transfer with {to.Name}");
             }
 
-            var youngFrom = from is PlayerMobile mobile && mobile.Young;
-            var youngTo = to is PlayerMobile playerMobile && playerMobile.Young;
-
-            if (youngFrom && !youngTo)
-            {
-                from.SendLocalizedMessage(502051); // As a young player, you may not transfer pets to older players.
-            }
-            else if (!youngFrom && youngTo)
+            if (IsYoungPlayerMismatchForTransfer(from, to))
             {
-                from.SendLocalizedMessage(502052); // As an older player, you may not transfer pets to young players.
+                // messages are now handled inside of IsYoungPlayerMismatchForTransfer method
             }
             else if (!m_Mobile.CanBeControlledBy(to))
             {
-                var args = $"{to.Name}\t{from.Name}\t ";
-
-                from.SendLocalizedMessage(
-                    1043248,
-                    args
-                ); // The pet refuses to be transferred because it will not obey ~1_NAME~.~3_BLANK~
-                to.SendLocalizedMessage(
-                    1043249,
-                    args
-                ); // The pet will not accept you as a master because it does not trust you.~3_BLANK~
+                SendTransferRefusalMessages(from, to, 1043248, 1043249);
+                // 1043248: The pet refuses to be transferred because it will not obey ~1_NAME~.~3_BLANK~
+                // 1043249: The pet will not accept you as a master because it does not trust you.~3_BLANK~
             }
             else if (!m_Mobile.CanBeControlledBy(from))
             {
-                var args = $"{to.Name}\t{from.Name}\t ";
-
-                from.SendLocalizedMessage(
-                    1043250,
-                    args
-                ); // The pet refuses to be transferred because it will not obey you sufficiently.~3_BLANK~
-                to.SendLocalizedMessage(
-                    1043251,
-                    args
-                ); // The pet will not accept you as a master because it does not trust ~2_NAME~.~3_BLANK~
+                SendTransferRefusalMessages(from, to, 1043250, 1043251);
+                // 1043250: The pet refuses to be transferred because it will not obey you sufficiently.~3_BLANK~
+                // 1043251: The pet will not accept you as a master because it does not trust ~2_NAME~.~3_BLANK~
             }
-            else if (TransferItem.IsInCombat(m_Mobile))
+            else if (IsInCombatState())
             {
-                from.SendMessage("You may not transfer a pet that has recently been in combat.");
-                to.SendMessage("The pet may not be transferred to you because it has recently been in combat.");
+                from.SendLocalizedMessage(1042561);
```

Comment:
1042561 is `Please dismount first.`

Reply:
Sorry, that is not what I meant. You changed the message from text to a localization number, but that isn't the correct number. In fact, I don't see that message localized.